marklogic-unit-test-client/src/main/java/com/marklogic/test/unit/JaxpServiceResponseUnmarshaller.java

```diff
@@ -67,10 +71,14 @@ public TestSuiteResult parseTestSuiteResult(String xml) {
 			NodeList resultNodes = testNode.getChildNodes();
 			String failureXml = null;
 			for (int j = 0; j < resultNodes.getLength(); j++) {
-				Element resultNode = (Element) resultNodes.item(j);
-				if ("fail".equals(resultNode.getAttribute("type"))) {
-					failureXml = toXml(resultNode);
-					break;
+				if (resultNodes.item(j).getNodeType() == ELEMENT_NODE) { // process only element node, ignore the rest
```

Comment:
I'm going to be picky here :) - this comment is restating what the code already states - i.e. that we only care about element nodes. It doesn't answer the question of "Why?", and the answer to that is provided by the bug description in the ticket. So I would do a comment like this:

```
// An XQuery test module may return additional nodes, such as text nodes, which should be
// ignored unless they are element nodes. 
```

Also, add _TYPE to "ELEMENT_NODE" to clarify that it's a node type, not a node.

Reply:
Added comment block